app/src/main/java/com/lightricks/feedexercise/data/FeedRepository.kt

```diff
@@ -1,9 +1,49 @@
 package com.lightricks.feedexercise.data
 
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Transformations
+import com.lightricks.feedexercise.database.FeedDao
+import com.lightricks.feedexercise.database.FeedDatabase
+import com.lightricks.feedexercise.database.FeedItemEntity
+import com.lightricks.feedexercise.network.FeedApiService
+import com.lightricks.feedexercise.network.TemplatesMetadata
+import io.reactivex.Completable
+
 /**
  * This is our data layer abstraction. Users of this class don't need to know
  * where the data actually comes from (network, database or somewhere else).
  */
-class FeedRepository {
-    //todo: implement
-}
\ No newline at end of file
+class FeedRepository(db:FeedDatabase, private val service: FeedApiService) {
```

Comment:
Can you make db a `private val` ?

Reply:
I can, but I purposely didn't not save it's reference as It's not needed once I have the DAO.
I fixed it, but I would love an explanation for why it is needed.

Reply:
I guess it's not necessary.